src/colvar_rotation_derivative.h

```diff
@@ -0,0 +1,604 @@
+                      tmp_Q0Q0_L[1][1][2] * ds[1][2] +
+                      tmp_Q0Q0_L[1][1][3] * ds[1][3] +
+                      tmp_Q0Q0_L[1][2][0] * ds[2][0] +
+                      tmp_Q0Q0_L[1][2][1] * ds[2][1] +
+                      tmp_Q0Q0_L[1][2][2] * ds[2][2] +
+                      tmp_Q0Q0_L[1][2][3] * ds[2][3] +
+                      tmp_Q0Q0_L[1][3][0] * ds[3][0] +
+                      tmp_Q0Q0_L[1][3][1] * ds[3][1] +
+                      tmp_Q0Q0_L[1][3][2] * ds[3][2] +
+                      tmp_Q0Q0_L[1][3][3] * ds[3][3];
+
+      (*dq0_out)[2] = tmp_Q0Q0_L[2][0][0] * ds[0][0] +
+                      tmp_Q0Q0_L[2][0][1] * ds[0][1] +
+                      tmp_Q0Q0_L[2][0][2] * ds[0][2] +
+                      tmp_Q0Q0_L[2][0][3] * ds[0][3] +
+                      tmp_Q0Q0_L[2][1][0] * ds[1][0] +
+                      tmp_Q0Q0_L[2][1][1] * ds[1][1] +
+                      tmp_Q0Q0_L[2][1][2] * ds[1][2] +
+                      tmp_Q0Q0_L[2][1][3] * ds[1][3] +
+                      tmp_Q0Q0_L[2][2][0] * ds[2][0] +
+                      tmp_Q0Q0_L[2][2][1] * ds[2][1] +
+                      tmp_Q0Q0_L[2][2][2] * ds[2][2] +
+                      tmp_Q0Q0_L[2][2][3] * ds[2][3] +
+                      tmp_Q0Q0_L[2][3][0] * ds[3][0] +
+                      tmp_Q0Q0_L[2][3][1] * ds[3][1] +
+                      tmp_Q0Q0_L[2][3][2] * ds[3][2] +
+                      tmp_Q0Q0_L[2][3][3] * ds[3][3];
+
+      (*dq0_out)[3] = tmp_Q0Q0_L[3][0][0] * ds[0][0] +
+                      tmp_Q0Q0_L[3][0][1] * ds[0][1] +
+                      tmp_Q0Q0_L[3][0][2] * ds[0][2] +
+                      tmp_Q0Q0_L[3][0][3] * ds[0][3] +
+                      tmp_Q0Q0_L[3][1][0] * ds[1][0] +
+                      tmp_Q0Q0_L[3][1][1] * ds[1][1] +
+                      tmp_Q0Q0_L[3][1][2] * ds[1][2] +
+                      tmp_Q0Q0_L[3][1][3] * ds[1][3] +
+                      tmp_Q0Q0_L[3][2][0] * ds[2][0] +
+                      tmp_Q0Q0_L[3][2][1] * ds[2][1] +
+                      tmp_Q0Q0_L[3][2][2] * ds[2][2] +
+                      tmp_Q0Q0_L[3][2][3] * ds[2][3] +
+                      tmp_Q0Q0_L[3][3][0] * ds[3][0] +
+                      tmp_Q0Q0_L[3][3][1] * ds[3][1] +
+                      tmp_Q0Q0_L[3][3][2] * ds[3][2] +
+                      tmp_Q0Q0_L[3][3][3] * ds[3][3];
+    }
+  }
+  /*! @brief Calculate the derivatives of S, the leading eigenvalue L and
+    *         the leading eigenvector Q with respect to `m_pos1`
+    *  @param[in]  ia        The index the of atom
+    *  @param[out] dl0_1_out The output of derivative of L with respect to
+    *                        ia-th atom of group 1
+    *  @param[out] dq0_1_out The output of derivative of Q with respect to
+    *                        ia-th atom of group 1
+    *  @param[out] ds_1_out  The output of derivative of overlap matrix S with
+    *                        respect to ia-th atom of group 1
+    */
+  void calc_derivative_to_group1(
```

Comment:
The full expression would be "derivative with respect to", which I would abbreviate as: `calc_derivative_wrt_group*`